modules/core/src/persistence_json.cpp

```diff
@@ -858,8 +858,6 @@ class JSONParser : public FileStorageParser
             CV_PARSE_ERROR_CPP( "left-brace of top level is missing" );
         }
 
-        if( !ptr || !*ptr )
-            CV_PARSE_ERROR_CPP( "Unexpected End-Of-File" );
```

Comment:
I believe `ptr` should be reassigned above instead:
```
ptr = parseMap(ptr, ...);
...
ptr = parseSeq(ptr, ...);
```

Reply:
I believe, those 2 lines just left out by mistake after refactoring. If `ptr` is reassigned, all the tests fail. Simply remove them.